rails_exercises/ecommerce_front/admin/app/controllers/admin/feature_products_controller.rb

```diff
@@ -0,0 +1,57 @@
+require_dependency "admin/application_controller"
+
+module Admin
+  class FeatureProductsController < ApplicationController
+    before_filter :set_feature_product, only: [:edit, :update, :destroy]
+
+    def index
+      @feature_products = FeatureProduct.all
+    end
+
+    def new
+      @feature_product = FeatureProduct.new
+    end
+
+    def create
+      @feature_product = FeatureProduct.new(feature_product_params)
+
+      if @feature_product.save
+        redirect_to action: 'index'
+      else
+        render :new
+      end
+    end
+
+    def edit; end
+
+    def update
+      @feature_product.update_attributes(feature_product_params)
+
+      if @feature_product.save
+        redirect_to action: 'index'
+      else
+        render :new
+      end
+    end
+
+    def delete_confirmation
+      @feature_product = FeatureProduct.find(params[:feature_product_id])
+    end
+
+    def destroy
+      FeatureProduct.destroy(params[:id])
+
+      redirect_to action: 'index'
+    end
+
+    private
+
+    def set_feature_product
+      @feature_product = FeatureProduct.find(params[:id])
+    end
+
+    def feature_product_params
+      params[:feature_product]
```

Comment:
It is safer to whitelist the allowed parameters explicitly here:
```
params[:feature_product].slice(:id, :name, ...)
```
Rails 4 has [strong parameters](http://guides.rubyonrails.org/v4.0/action_controller_overview.html#strong-parameters) for that.

Reply:
Yes, I've been thinking about that. One link I consulted was this one - https://stackoverflow.com/questions/13852534/strong-parameters-in-rails-3-2-8 Since we use Rails 3.22.2, I decided to skip whitelisting. Is there a way I can use it without adding external gems?